apis/python/src/tiledbsoma/factory.py

```diff
@@ -56,40 +56,44 @@ def _construct_member(
     try:
         if object_type == "array":
             with tiledb.open(member_uri, ctx=ctx) as A:
-                class_name = A.meta[SOMA_OBJECT_TYPE_METADATA_KEY]
+                spec_name = A.meta[SOMA_OBJECT_TYPE_METADATA_KEY]
         elif object_type == "group":
             with tiledb.Group(member_uri, mode="r", ctx=ctx) as G:
-                class_name = G.meta[SOMA_OBJECT_TYPE_METADATA_KEY]
+                spec_name = G.meta[SOMA_OBJECT_TYPE_METADATA_KEY]
         else:
             return None
 
     except tiledb.TileDBError:
         return None
 
-    assert class_name is not None
+    assert spec_name is not None
+    if spec_name not in SPEC_NAMES_TO_CLASS_NAMES:
+        raise Exception(f'name "{spec_name}" unrecognized')
```

Comment:
this should be an error derived from SOMAError, right?  In any case, it is generally a bad idea to raise naked Exception instances

Reply:
@bkmartinjr good call!! :)

There are more such in the repo so I filed #467 as a follow-on

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/464/commits/7cf886146a80027630255943d984b71bd38f8b4c